src/main/kotlin/org/jetbrains/research/libsl/visitors/TypeResolver.kt

```diff
@@ -109,4 +110,15 @@ class TypeResolver(
 
         return name to typeRef
     }
+
+    override fun visitBitShiftOp(ctx: LibSLParser.BitShiftOpContext) {
+        val token = (ctx.getChild(0) as TerminalNode).symbol
+        val firstIndex = token.startIndex
+        for (i in 1 until ctx.childCount) {
```

Comment:
This logic is too complex

Reply:
Check for shift operators is now simple, this code was removed